lisa/src/main/java/it/unive/lisa/analysis/nonrelational/heap/NonRelationalHeapDomain.java

```diff
@@ -17,4 +19,46 @@
  */
 public interface NonRelationalHeapDomain<T extends NonRelationalHeapDomain<T>>
 		extends NonRelationalDomain<T, SymbolicExpression, HeapEnvironment<T>>, HeapSemanticOperation {
+
+	@Override
+	public default HeapEnvironment<T> assume(HeapEnvironment<T> environment, SymbolicExpression expression,
+			ProgramPoint pp) throws SemanticException {
+		return environment;
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public default T glb(T other) throws SemanticException {
+		if (other == null || this.isBottom() || other.isTop() || this == other || this.equals(other)
+				|| this.lessOrEqual(other))
+			return (T) this;
+
+		if (other.isBottom() || this.isTop() || other.lessOrEqual((T) this))
+			return (T) other;
+
+		return glbAux(other);
+	}
+
+	/**
+	 * Performs the greatest lower bound operation between this domain element
+	 * and {@code other}, assuming that base cases have already been handled. In
+	 * particular, it is guaranteed that:
+	 * <ul>
+	 * <li>{@code other} is not {@code null}</li>
+	 * <li>{@code other} is neither <i>top</i> nor <i>bottom</i></li>
+	 * <li>{@code this} is neither <i>top</i> nor <i>bottom</i></li>
+	 * <li>{@code this} and {@code other} are not the same object (according
+	 * both to {@code ==} and to {@link Object#equals(Object)})</li>
+	 * <li>{@code this} and {@code other} are not comparable (according to
+	 * {@link Lattice#lessOrEqual(Lattice)})</li>
+	 * </ul>
+	 * The default implementation returns {@link Lattice#bottom()}
+	 * 
+	 * @param other the other domain element
+	 * 
+	 * @return the greatest lower bound between this domain element and other
+	 */
+	public default T glbAux(T other) {
```

Comment:
This should definitely not be here: it is supposed to be an auxiliary method exposed only to subclasses.